--- a/scalelib/src/file/scale_file_netcdf.c.orig	2025-12-17 14:29:44.000000000 +0900
+++ b/scalelib/src/file/scale_file_netcdf.c	2025-12-17 14:30:05.000000000 +0900
@@ -1753,7 +1753,7 @@
     if ( strlen(desc)   >0 ) CHECK_ERROR( nc_put_att_text(ncid, varid, "long_name", strlen(desc), desc) )
     if ( strlen(units)  >0 ) CHECK_ERROR( nc_put_att_text(ncid, varid, "units", strlen(units), units) )
     if ( strlen(stdname)>0 ) CHECK_ERROR( nc_put_att_text(ncid, varid, "standard_name", strlen(stdname), stdname) )
-    CHECK_ERROR( nc_put_att_double(ncid, varid, _FillValue, xtype, 1, &rmiss) )
+    CHECK_ERROR( nc_put_att_double(ncid, varid, "_FillValue", xtype, 1, &rmiss) )
     CHECK_ERROR( nc_put_att_double(ncid, varid, "missing_value", xtype, 1, &rmiss) )
   }
   if ( has_assoc ) {
