diff --git a/src/base/openmp_trace.c b/src/base/openmp_trace.c
index 84bd46ba09..3d8303a5bc 100644
--- a/src/base/openmp_trace.c
+++ b/src/base/openmp_trace.c
@@ -27,9 +27,9 @@ int openmp_trace_issues(void) { /* routine is exposed in Fortran interface */
  * __clang__: omp and iomp/icx, __INTEL_COMPILER: iomp/icc
  * __INTEL_LLVM_COMPILER: already covered by __clang__
  */
-#if (defined(__clang__) || defined(__INTEL_COMPILER))
-#include <omp-tools.h>
-#else
+//#if (defined(__clang__) || defined(__INTEL_COMPILER))
+//#include <omp-tools.h>
+//#else
 typedef struct ompt_frame_t ompt_frame_t;
 typedef void *ompt_initialize_t;
 typedef void *ompt_finalize_t;
@@ -80,7 +80,7 @@ typedef int (*ompt_get_parallel_info_t)(int, ompt_data_t **, int *);
 typedef ompt_interface_fn_t (*ompt_function_lookup_t)(const char *);
 typedef ompt_set_result_t (*ompt_set_callback_t)(ompt_callbacks_t,
                                                  ompt_callback_t);
-#endif
+//#endif
 
 #if !defined(_WIN32) && !defined(__CYGWIN__) && !defined(OPENMP_TRACE_SYMBOL)
 #define OPENMP_TRACE_SYMBOL
diff --git a/src/fm/cp_fm_elpa.F b/src/fm/cp_fm_elpa.F
index c16adb2aaa..aa2a367778 100644
--- a/src/fm/cp_fm_elpa.F
+++ b/src/fm/cp_fm_elpa.F
@@ -187,7 +187,9 @@ CONTAINS
 ! **************************************************************************************************
    SUBROUTINE finalize_elpa_library()
 #if defined(__ELPA)
-      CALL elpa_uninit()
+      USE ISO_C_BINDING, ONLY: C_INT
+      INTEGER(KIND=C_INT)                 :: error_elpa
+      CALL elpa_uninit(error_elpa)
 #else
       CPABORT("Finalization of ELPA library requested but not enabled during build")
 #endif
@@ -322,6 +324,8 @@ CONTAINS
 ! **************************************************************************************************
    SUBROUTINE cp_fm_diag_elpa_base(matrix, eigenvectors, eigenvalues, rdinfo)
 
+      USE ISO_C_BINDING, ONLY: C_INT
+      INTEGER(KIND=C_INT)                 :: error_elpa
       TYPE(cp_fm_type), INTENT(IN)                       :: matrix, eigenvectors
       REAL(KIND=dp), DIMENSION(:), INTENT(OUT)           :: eigenvalues
       TYPE(cp_fm_redistribute_info), INTENT(IN)          :: rdinfo
@@ -471,7 +475,7 @@ CONTAINS
       ! the full eigenvalues vector is needed
       ALLOCATE (eval(n))
 
-      elpa_obj => elpa_allocate()
+      elpa_obj => elpa_allocate(error_elpa)
 
       CALL elpa_obj%set("na", n, success)
       CPASSERT(success == elpa_ok)
diff --git a/src/grid/cpu/grid_cpu_collint.h b/src/grid/cpu/grid_cpu_collint.h
index 2f3aeaebd6..aaf8185130 100644
--- a/src/grid/cpu/grid_cpu_collint.h
+++ b/src/grid/cpu/grid_cpu_collint.h
@@ -45,19 +45,22 @@ ortho_cx_to_grid_scalar(const int lp, const int cmax, const int i,
 
 #if (GRID_DO_COLLOCATE)
   // collocate
-  double reg[4] = {0.0, 0.0, 0.0, 0.0};
-#pragma omp simd reduction(+ : reg)
+  double reg0 = 0.0;
+  double reg1 = 0.0;
+  double reg2 = 0.0;
+  double reg3 = 0.0;
+#pragma omp simd reduction(+:reg0, reg1, reg2, reg3)
   for (int lxp = 0; lxp <= lp; lxp++) {
     const double p = pol[0][lxp][i + cmax];
-    reg[0] += cx[lxp * 4 + 0] * p;
-    reg[1] += cx[lxp * 4 + 1] * p;
-    reg[2] += cx[lxp * 4 + 2] * p;
-    reg[3] += cx[lxp * 4 + 3] * p;
+    reg0 += cx[lxp * 4 + 0] * p;
+    reg1 += cx[lxp * 4 + 1] * p;
+    reg2 += cx[lxp * 4 + 2] * p;
+    reg3 += cx[lxp * 4 + 3] * p;
   }
-  *grid_0 += reg[0];
-  *grid_1 += reg[1];
-  *grid_2 += reg[2];
-  *grid_3 += reg[3];
+  *grid_0 += reg0;
+  *grid_1 += reg1;
+  *grid_2 += reg2;
+  *grid_3 += reg3;
 
 #else
   // integrate
diff --git a/src/grid/cpu/grid_cpu_collocate.c b/src/grid/cpu/grid_cpu_collocate.c
index 2dd1f4894e..925ef4c81f 100644
--- a/src/grid/cpu/grid_cpu_collocate.c
+++ b/src/grid/cpu/grid_cpu_collocate.c
@@ -44,7 +44,7 @@ static void write_task_file(
   char filename[100];
   snprintf(filename, sizeof(filename), "grid_collocate_%05i.task", my_number);
 
-  const int D = DBL_DECIMAL_DIG;
+  const int D = DECIMAL_DIG;
   FILE *fp = fopen(filename, "w+");
   fprintf(fp, "#Grid task v10\n");
   fprintf(fp, "orthorhombic %i\n", orthorhombic);
